### ISSUE TITLE
train_ppo_ray，不用vllm训练会出现推理乱码的情况

### ISSUE THREAD

**Issue by zuxinqi** [OpenRLHF/OpenRLHF#956]
第一次的推理样例有很多的乱码，发现action_log_probs的值，使用vllm\packing_samples与不使用vllm\packing_samples差别很大，使用vllm\packing_samples不出现乱码，我用的是lora训练，用vllm\packing_samples后面更新会报错。在非lora训练上也尝试了，不用vllm\packing_samples第一次推理依旧出现乱码

训练的参数为
ray job submit --address="http://127.0.0.1:8265" \
   --runtime-env-json='{"working_dir": "/data7/rlhf_test/openrlhf"}' \
   -- python3 -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 1 \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 1 \
   --vllm_num_engines 0 \
   --vllm_tensor_parallel_size 1 \
   --pretrain /data7/llm_base_model/ \
   --remote_rm_url http://192.168.100.101:5000/get_score
   --save_path /data7/rlhf_test/openrlhf/ \
   --micro_train_batch_size 8 \
   --train_batch_size 128 \
   --micro_rollout_batch_size 8 \
   --rollout_batch_size 32 \
   --lora_rank 8 \
   --lora_alpha 16 \
   --n_samples_per_prompt 1 \
   --max_epochs 1 \
   --prompt_max_len 1024 \
   --max_samples 100000 \
   --generate_max_len 1024 \
   --advantage_estimator reinforce \
   --bf16 \
   --actor_learning_rate 4e-7 \
   --init_kl_coef 1e-4 \
   --prompt_data /data7/rlhf_test/rlhf_data_0401/ \
   --input_key chat \
   --apply_chat_template \
   --normalize_reward \
   --gradient_checkpointing \
   --save_steps -1 \
   --ckpt_path /data7/ \
   --use_tensorboard /data7/rlhf_test/openrlhf/tensorboard \
   > run_reinforcelora_0410.log 2>&1 &

**Comment by AIzealotwu** [comment]
我出现了这个问题，没法按照之前sft模型规定的格式做输出，自定义了规则奖励

**Comment by AIzealotwu** [comment]
> 第一次的推理样例有很多的乱码，发现action_log_probs的值，使用vllm\packing_samples与不使用vllm\packing_samples差别很大，使用vllm\packing_samples不出现乱码，我用的是lora训练，用vllm\packing_samples后面更新会报错。在非lora训练上也尝试了，不用vllm\packing_samples第一次推理依旧出现乱码
> 
> 训练的参数为 ray job submit --address="http://127.0.0.1:8265" --runtime-env-json='{"working_dir": "/data7/rlhf_test/openrlhf"}' -- python3 -m openrlhf.cli.train_ppo_ray --ref_num_nodes 1 --ref_num_gpus_per_node 1 --actor_num_nodes 1 --actor_num_gpus_per_node 1 --vllm_num_engines 0 --vllm_tensor_parallel_size 1 --pretrain /data7/llm_base_model/ --remote_rm_url http://192.168.100.101:5000/get_score --save_path /data7/rlhf_test/openrlhf/ --micro_train_batch_size 8 --train_batch_size 128 --micro_rollout_batch_size 8 --rollout_batch_size 32 --lora_rank 8 --lora_alpha 16 --n_samples_per_prompt 1 --max_epochs 1 --prompt_max_len 1024 --max_samples 100000 --generate_max_len 1024 --advantage_estimator reinforce --bf16 --actor_learning_rate 4e-7 --init_kl_coef 1e-4 --prompt_data /data7/rlhf_test/rlhf_data_0401/ --input_key chat --apply_chat_template --normalize_reward --gradient_checkpointing --save_steps -1 --ckpt_path /data7/ --use_tensorboard /data7/rlhf_test/openrlhf/tensorboard \
> 
> > run_reinforcelora_0410.log 2>&1 &

请问你有打印prompts和queries吗，有没发现其中有问题

**Comment by pizts** [comment]
请问有解决了吗
